app/models/mark.rb

```diff
@@ -18,6 +18,27 @@ class Mark < ApplicationRecord
   validates_uniqueness_of :markable_id,
                           scope: [:result_id, :markable_type]
 
+  validates_inclusion_of :override, in: [true, false]
+
+  def calculate_deduction
+    return 0 if self.markable_type != 'FlexibleCriterion' || self.override?
+
+    # use joins to fix this statement to be query
+    deductive_annotations = self.result
```

Comment:
You don't need a variable here, you can return the result of the query directly.